From 660a2e065adbbb057c47b34712598a6f29d87305 Mon Sep 17 00:00:00 2001
From: Andreas Mohr <and@gmx.li>
Date: Sat, 21 May 2016 14:24:13 +0000
Subject: [PATCH] Save user abort request at generic search function

Save user abort request as MC_SEARCH_CB_ABORT error status.

It is a bit fuzzy here because of ret = MC_SEARCH_CB_ABORT initial default.

Signed-off-by: Andreas Mohr <and@gmx.li>
---
 lib/search/regex.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/lib/search/regex.c b/lib/search/regex.c
index a172640..53e2729 100644
--- a/lib/search/regex.c
+++ b/lib/search/regex.c
@@ -901,7 +901,10 @@ mc_search__run_regex (mc_search_t * lc_mc_search, const void *user_data,
                 ret = lc_mc_search->search_fn (user_data, current_pos, &current_chr);
 
                 if (ret == MC_SEARCH_CB_ABORT)
+                {
+                    lc_mc_search->error = MC_SEARCH_E_ABORT;
                     break;
+                }
 
                 if (ret == MC_SEARCH_CB_INVALID)
                     continue;
@@ -967,7 +970,10 @@ mc_search__run_regex (mc_search_t * lc_mc_search, const void *user_data,
 
         if ((lc_mc_search->update_fn != NULL) &&
             ((lc_mc_search->update_fn) (user_data, current_pos) == MC_SEARCH_CB_ABORT))
+        {
+            lc_mc_search->error = MC_SEARCH_E_ABORT;
             ret = MC_SEARCH_CB_ABORT;
+        }
 
         if (ret == MC_SEARCH_CB_ABORT)
             break;
@@ -975,7 +981,8 @@ mc_search__run_regex (mc_search_t * lc_mc_search, const void *user_data,
 
     g_string_free (lc_mc_search->regex_buffer, TRUE);
     lc_mc_search->regex_buffer = NULL;
-    lc_mc_search->error = MC_SEARCH_E_NOTFOUND;
+    if (lc_mc_search->error != MC_SEARCH_E_ABORT)
+        lc_mc_search->error = MC_SEARCH_E_NOTFOUND;
 
     if (ret != MC_SEARCH_CB_ABORT)
         lc_mc_search->error_str = g_strdup (_(STR_E_NOTFOUND));
